protocols/mdns/src/behaviour.rs

```diff
@@ -54,7 +54,11 @@ pub struct MdnsConfig {
     /// peer joins the network. Receiving an mdns packet resets the timer
     /// preventing unnecessary traffic.
     pub query_interval: Duration,
-    /// IP address for multicast.
+    /// Internet protocol version (v4 or v6) to use.
+    ///
+    /// Note that the provided IP address itself is ignored and instead the
+    /// unspecified address (`0.0.0.0` or `::`) of the corresponding version is
+    /// used.
```

Comment:
Instead of including this note, we could make the fields of this config private and include constructors like:

```rust
MdnsConfig::ipv4().with_ttl().with_query_interval();
MdnsConfig::ipv6().with_ttl().with_query_interval();
```

Thoughts?

Reply:
The comment change, which I suggested, is actually wrong. See https://github.com/libp2p/rust-libp2p/pull/2359/files#r758241261. The address is actually used, e.g. when joining the multicast group. Sorry, my bad.